Firestore/core/src/util/executor_std.cc

```diff
@@ -131,8 +143,15 @@ void ExecutorStd::OnCompletion(Task*) {
 }
 
 void ExecutorStd::Cancel(const Id operation_id) {
-  auto removed = state_->schedule_.RemoveIf(
-      [operation_id](const Task& t) { return t.id() == operation_id; });
+  Task* removed = nullptr;
+  {
+    std::lock_guard<std::mutex> lock(mutex_);
```

Comment:
`schedule_` already has an internal mutex. Is this additional mutex to protect `state_` pointer itself?

Reply:
Correct.

We could probably remove the internal mutex on schedule (though it makes `PopBlocking` something that will require a `unique_lock` as an argument.

Reply:
I actually prefer `Schedule` being self-contained (efficiency concerns may of course override that, but I don't think we're at that point yet), so I'm certainly not pushing for something like that.